tests/planner/test_SnapPlanner.cpp

```diff
@@ -51,43 +57,80 @@ class SnapPlannerTest : public ::testing::Test
   shared_ptr<PassingConstraint> passingConstraint;
   shared_ptr<FailingConstraint> failingConstraint;
   shared_ptr<GeodesicInterpolator> interpolator;
-  aikido::planner::PlanningResult planningResult;
+  SnapConfigurationToConfigurationPlanner::Result planningResult;
```

Comment:
What's the difference between `SnapConfigurationToConfigurationPlanner::Result` and `Planner::Result`? Maybe it'd be easier if we can always use `Planner::Result` so we don't have to change our result type if we decide we want to use a different planner? What would happen for a meta-planner?

Reply:
They are identical. Note that `SnapConfigurationToConfigurationPlanner` doesn't actually typedefs `Result`, but it just `Planner::Result`. Using `[X]Planner::Result` would be a safe way because it will always use the right result type.

Planners could have different types of information so it would be more flexible to define different result type when necessary.

Maybe we need to introduce `Planner::createResult()` to be able to create the right result from a base planner pointer like we did for `Outcome`.

Reply:
Sounds good. Let's merge this PR first and add that change later.